modules/backup-s3/client.go

```diff
@@ -61,37 +66,85 @@ func newClient(config *clientConfig, logger logrus.FieldLogger, dataPath string)
 	if err != nil {
 		return nil, errors.Wrap(err, "create client")
 	}
-	return &s3Client{client, config, logger, dataPath}, nil
+	return &s3Client{client, config, logger, dataPath, bucket, path, creds, region}, nil
+}
+
+func tempClient(currentClient *s3Client, overrideBucket, overridePath string, ctx context.Context) (*s3Client, error) {
+	AwsAccessKey := modulecomponents.GetValueFromContext(ctx, "X-AWS-ACCESS-KEY")
+	xAwsSecretKey := modulecomponents.GetValueFromContext(ctx, "X-AWS-SECRET-KEY")
+	xAwsSessionToken := modulecomponents.GetValueFromContext(ctx, "X-AWS-SESSION-TOKEN")
+	if overrideBucket != "" {
+		config := currentClient.config
+		creds := currentClient.creds
+		if AwsAccessKey != "" && xAwsSecretKey != "" {
+			creds = credentials.NewStaticV4(AwsAccessKey, xAwsSecretKey, xAwsSessionToken)
+		}
+
+		client, err := minio.New(config.Endpoint, &minio.Options{
+			Creds:  creds,
+			Region: currentClient.region,
+			Secure: config.UseSSL,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "create minio client")
+		}
+		return &s3Client{client, config, currentClient.logger, currentClient.dataPath, overrideBucket, overridePath, creds, currentClient.region}, nil
+	} else {
+		return currentClient, nil
```

Comment:
here in this case if somebody is not overriding bucket but wants to provide credentials using headers we will not return a client to him that respects those creds provided with headers, can we take this scenario also into account?

Reply:
done